vendor: point vishvananda/netlink back to upstream

borkmann opened the pull request:
Given PR https://github.com/vishvananda/netlink/pull/498 has been
merged officially, move the repo back to upstream. Final follow-up
to 515654b58beb ("vendor: fix stack corruption from retrieving
veth peer index"). Re-performed same update as in 515654b58beb.

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9606)
<!-- Reviewable:end -->

aanm reviewed (changes requested):
from master branch, please apply:
```patch
diff --git a/go.mod b/go.mod
index 4d2dc94e8..267428368 100644
--- a/go.mod
+++ b/go.mod
@@ -497,7 +497,7 @@ replace (
        github.com/ugorji/go => github.com/ugorji/go v1.1.4
        github.com/urfave/cli => github.com/urfave/cli v1.20.0
        github.com/urfave/negroni => github.com/urfave/negroni v1.0.0
-       github.com/vishvananda/netlink => github.com/cilium/netlink v1.0.1-0.20191111102915-47ae870a10ed
+       github.com/vishvananda/netlink => github.com/vishvananda/netlink v1.0.1-0.20191113183427-d71301a47b60
        github.com/vishvananda/netns => github.com/vishvananda/netns v0.0.0-20190625233234-7109fa855b0f
        github.com/vmware/govmomi => github.com/vmware/govmomi v0.20.1
        github.com/xiang90/probing => github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2
```
and then run:
`./contrib/go-mod/update-vendor.sh`

borkmann committed:
vendor: point vishvananda/netlink back to upstream

Given PR https://github.com/vishvananda/netlink/pull/498 has been
merged officially, move the repo back to upstream. Final follow-up
to 515654b58beb ("vendor: fix stack corruption from retrieving
veth peer index"). Re-performed same update as in 515654b58beb.

Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
